tests/ui/parser/misspelled-keywords/impl-trait.stderr

```diff
@@ -0,0 +1,17 @@
+error: expected one of `(`, `+`, `,`, `::`, `<`, `=`, or `>`, found `Debug`
+  --> $DIR/impl-trait.rs:1:18
+   |
+LL | fn code<T: impll Debug>() -> u8 {}
+   |                  ^^^^^ expected one of 7 possible tokens
+   |
+help: there is a keyword `impl` with a similar name
+   |
+LL | fn code<T: impl Debug>() -> u8 {}
+   |            ~~~~
+help: you might have meant to end the type parameters here
```

Comment:
can we remove the later errors if we report similar keyword suggestion? I'm not sure, in this case, we should remove it.

Reply:
Yeah, I agree. Other suggestions are meaningless once a misspelled keyword is detected.

But, currently there is no way to prevent new suggestions from being added to a `rustc_errors::Diag`. `disable_suggestions()` is the closest but it gets rid of all suggestions before and after the call.

So implementing a way to freeze the suggestions list would require changing the type of `suggestions` field in `rustc_errors::diagnostic::DiagInner` and this would touch a number of different files. 

Should I implement it in this PR itself or create a separate PR and `r?` you?

Reply:
seems better in another PR.